internal/rpc/server/ospkg/server.go

```diff
@@ -0,0 +1,41 @@
+package ospkg
+
+import (
+	"context"
+
+	"github.com/google/wire"
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/trivy/internal/rpc"
+	detector "github.com/aquasecurity/trivy/pkg/detector/ospkg"
+	"github.com/aquasecurity/trivy/pkg/log"
+	"github.com/aquasecurity/trivy/pkg/vulnerability"
+	proto "github.com/aquasecurity/trivy/rpc/detector"
+)
+
+var SuperSet = wire.NewSet(
+	detector.SuperSet,
+	vulnerability.SuperSet,
+	NewServer,
+)
+
+type Server struct {
+	detector   detector.Operation
+	vulnClient vulnerability.Operation
+}
+
+func NewServer(detector detector.Operation, vulnClient vulnerability.Operation) *Server {
+	return &Server{detector: detector, vulnClient: vulnClient}
+}
+
+func (s *Server) Detect(ctx context.Context, req *proto.OSDetectRequest) (res *proto.DetectResponse, err error) {
+	vulns, err := s.detector.Detect(req.OsFamily, req.OsName, rpc.ConvertFromRpcPkgs(req.Packages))
+	if err != nil {
+		log.Logger.Warn(err)
```

Comment:
I'd skip this log and stick to the rule; either log or propagate error to avoid duplicates

Reply:
The same reason. If it is confusing, we can add a comment.
https://github.com/aquasecurity/trivy/pull/303#discussion_r355524524